### ISSUE TITLE
add asyn functionality similar to parallel

### ISSUE THREAD

**Issue by 7fe** [caolan/async#55]
It we be nice to have a central library containing some solid core **Async** functionality.

Although the current `async.parallel` function is useful, it fails to respect function order giving useless results.

``` js
async.parallel([
    function(callback){
        setTimeout(function(){
            callback(null, 'one');
        }, 200);
    },
    function(callback){
        setTimeout(function(){
            callback(null, 'two');
        }, 100);
    },
],
// optional callback
function(err, results){
    // in this case, the results array will equal ['two','one']
    // because the functions were run in parallel and the second
    // function had a shorter timeout before calling the callback
    // because the results depend on 
});
```

I would suggest merging similar functionality of [asyn](https://github.com/limeblack/asyn) into `async'.

It provides a very clean syntax for creating `async` code that runs in parallel.

``` js
asyn
  (fs.writeFile)('foo','bar',asyn)// func 0
  (fs.readFile)('server.js',asyn)              // func 1
  .end(function(asyn){
    /*
     * Log the second argument (a buffer) of the 2nd func
     */
    console.log(String(asyn[1][1]));
  })

```

If it doesn't seem fit for `async`, it could remain separate, but it adds missing and beneficial functionality.
